build.gradle

```diff
@@ -118,6 +120,24 @@ subprojects {
         assertsEnabled = true
       }
     }
+
+    options.compilerArgs += [
+        // enable all warnings
+        '-Xlint:all',
+        // hide warning: No processor claimed any of these annotations: dagger.*,org.junit.Test
+        '-Xlint:-processing',
+        // hide warning: MethodParameters attribute introduced in version 52.0 class files is ignored in version 51.0 class files
+        '-Xlint:-classfile',
+        // hide warning: auxiliary class C in full/path/to/C.java should not be accessed from outside its own source file
```

Comment:
What triggers all of these? I'm less interested in the description since that can be Googled but more interested in what causes them and why it's safe to ignore. Similar to an `@Ignore` for a test. The 'processing' one has a good justification but not these last two.

Reply:
All 3 are the warning messages, not explanation. I'll research and explain them in the context of this project.